packages/integrations/src/awslambda.ts

```diff
@@ -0,0 +1,230 @@
+     * @param scope  - holds additional event information
+     * @hidden
+     */
+    const setCloudwatchLogsData = (scope: Scope): void => {
+      scope.setExtra('cloudwatch logs', {
+        log_group: this._awsContext.logGroupName,
+        log_stream: this._awsContext.logStreamName,
+        url: cloudwatchUrl(),
+      });
+    };
+
+    /**
+     * This function sets tags which are displayed in Sentry Dashboard
+     * @param scope  - holds additional event information
+     * @hidden
+     */
+
+    const setTags = (scope: Scope): void => {
+      scope.setTag('runtime', `node${global.process.version}`);
+      scope.setTag('transaction', this._awsContext.functionName);
+      scope.setTag('runtime.name', 'node');
+      scope.setTag('server_name', processEnv._AWS_XRAY_DAEMON_ADDRESS);
+      scope.setTag('url', `awslambda:///${this._awsContext.functionName}`);
+    };
+
+    /**
+     * setting parameters in scope which will be displayed as additional data in Sentry dashboard
+     * @hidden
+     */
+    const setParameters = (): void => {
+      // setting parameters in scope which will be displayed as additional data in Sentry dashboard
+      hub.configureScope((scope: Scope) => {
+        setTags(scope);
+        // runtime
+        setAdditionalRuntimeData(scope);
+        // setting the lambda parameters
+        setAdditionalLambdaParameters(scope);
+        // setting the cloudwatch logs parameter
+        setCloudwatchLogsData(scope);
+        // setting the sys.argv parameter
+        scope.setExtra('sys.argv', process.argv);
+      });
+    };
+
+    // timeout warning buffer for timeout error
+    const timeoutWarningBuffer: number = 1500;
+
+    /** check timeout flag and checking if configured Time In Milliseconds is greater than timeout Warning Buffer */
+    if (this._timeoutWarning === true && configuredTimeInMilliseconds > timeoutWarningBuffer) {
+      const configuredTimeInSec = Math.floor(configuredTimeInMilliseconds / 1000);
+      const configuredTimeInMilli = configuredTimeInSec * 1000;
+
+      /**
+       * This function is invoked when there is timeout error
+       * Here, we make sure the error has been captured by Sentry Dashboard
+       * and then re-raise the exception
+       * @param configuredTime  - configured time in seconds
+       * @hidden
+       */
+      const timeOutError = (configuredTime: number): void => {
+        setTimeout(() => {
+          /**
+           * setting parameters in scope which will be displayed as additional data in Sentry dashboard
+           */
+          setParameters();
+
+          const error = new Error(
+            `WARNING : Function is expected to get timed out. Configured timeout duration = ${configuredTimeInSec +
+              1} seconds.`,
+          );
+
+          /** capturing the exception and re-directing it to the Sentry Dashboard */
+          hub.captureException(error);
+          Sentry.flush(this._flushTime);
```

Comment:
Is AWS itself is waiting for unresolved promises? If not, then this `flush` will be kinda useless without any `await`.

Reply:
Also use `hub.flush()` instead, so we don't have any dependency on `@sentry/browser`

Reply:
We cant add async-await to the setTimeout function as it is changing the behavior.
Flush() method is inside getClient() function, so we use _hub.getClient().flush()_